corehq/apps/data_dictionary/util.py

```diff
@@ -303,3 +304,41 @@ def get_gps_properties(domain, case_type):
         case_type__name=case_type,
         data_type=CaseProperty.DataType.GPS,
     ).values_list('name', flat=True))
+
+
+def get_column_headings(row, valid_values, case_type=None):
+    column_headings = []
+    errors = []
+    for index, cell in enumerate(row, start=1):
+        if not cell.value:
+            if case_type:
+                errors.append(
+                    _("Column {} in case type {} has an empty header").format(index, case_type)
+                )
+            else:
+                errors.append(
+                    _("Column {} has an empty header").format(index)
+                )
+            continue
+
+        cell_value = cell.value.lower()
+        if cell_value in valid_values:
+            column_headings.append(valid_values[cell_value])
+        else:
+            if case_type:
+                errors.append(
+                    _("Invalid column \"{}\" in case type {}").format(cell.value, case_type)
+                )
+            else:
+                errors.append(
+                    _("Invalid column \"{}\"").format(cell.value)
+                )
+    return column_headings, errors
+
+
+def row_to_dict(row, column_headings, default_val=None):
```

Comment:
I think a name like `get_column_values_by_name` or something similar might describe this method better?
A few other ideas maybe:
- `map_row_values_to_column_names`
- `row_value_lookup_dict`

Reply:
I have replaced the name with `map_row_values_to_column_names` [05b3a44](https://github.com/dimagi/commcare-hq/pull/33259/commits/05b3a44d48b3ed7600c9cddc6cd86a1aeef07a32).